src/scenic/syntax/compiler.py

```diff
@@ -253,6 +253,44 @@ def visit_Require(self, node: s.Require):
             )
         )
 
+    def visit_RequireAlways(self, node: s.RequireAlways):
+        condition = self.visit(node.cond)
+        syntax_id = self._register_requirement_syntax(condition)
+        return ast.Expr(
```

Comment:
Since this same code is used for `require`, `require always`, `require eventually`, `record`, `record initial`, etc. with only the veneer function to call being different (and the optional probability in the case of `require`), I'd suggest putting it into a helper function.